components/log4j/org.wso2.carbon.analytics.shared.data.agents.log4j/src/main/java/org/wso2/carbon/analytics/shared/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -430,8 +433,18 @@ private void publishLogEvent(TenantDomainAwareLoggingEvent event) throws ParseEx
             if (event.getThrowableInformation() != null) {
                 arbitraryDataMap.put(columns[8], stacktrace);
             }
-            Event laEvent = new Event(streamDef, date.getTime(), null, null, new String[]{tenantDomain}, arbitraryDataMap);
-            dataPublisher.publish(laEvent);
+            Event logEvent = new Event(streamDef, date.getTime(), null, null, new String[]{tenantDomain},
+                    arbitraryDataMap);
+            if (dataPublisher != null) {
+                dataPublisher.publish(logEvent);
+            } else {
+                publisherInitializer();
```

Comment:
Rename accordingly in following commit : https://github.com/wso2/shared-analytics/pull/144/commits/16558450070256e23ae75c7d0c573d5a0db19638